File: src/main/java/com/mikea/gae/rx/RxHttpRequest.java
package com.mikea.gae.rx;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/**
 * @author mike.aizatsky@gmail.com
 */
public class RxHttpRequest {
    private final Rx rx;
    public final HttpServletRequest request;
    private final HttpServletResponse response;

    public RxHttpRequest(Rx rx, HttpServletRequest request, HttpServletResponse response) {
        this.rx = rx;
        this.request = request;
        this.response = response;
    }
}

File: src/main/scala/com/mikea/gae/rx/events/RxHttpRequest.scala
package com.mikea.gae.rx.events

import javax.servlet.http.{HttpServletResponse, HttpServletRequest}
import com.mikea.gae.rx.Rx

/**
 * @author mike.aizatsky@gmail.com
 */
class RxHttpRequest(val httpRequest: HttpServletRequest, val httpResponse: HttpServletResponse) extends RxEvent {
  private[rx] var processed : Boolean = false

  def this(request: RxHttpRequest) = this(request.httpRequest, request.httpResponse)
}